### PR TITLE
Add option to get remote channel id

### PR DESCRIPTION
**Pull request by obsd**

### CODE REVIEW

File: fastapi_websocket_rpc/rpc_channel.py [hunk 1]
```diff
@@ -156,6 +165,10 @@ def __init__(self, methods: RpcMethodsBase, socket, channel_id=None, default_res
     def context(self) -> Dict[str, Any]:
         return self._context
 
+    async def get_other_channel_id(self) -> str:
```
[the remark this answers is not in the document]
Comment (asafc) [reply fastapi_websocket_rpc/rpc_channel.py:168]: you have two methods with the signature `async def get_other_channel_id(self)`:
one here (line 168) and one in line 258.

i assume one of them should be `_get_other_channel_id` with underscore (i assume the one on 258) and that on connect should call `_get_other_channel_id`.

File: fastapi_websocket_rpc/rpc_channel.py [hunk 2]
```diff
@@ -234,8 +247,24 @@ async def on_handler_event(self, handlers, *args, **kwargs):
...
+        if self._sync_channel_id:
+            asyncio.create_task(self.get_other_channel_id())
         await self.on_handler_event(self._connect_handlers, self)
 
+    async def get_other_channel_id(self):
+        '''
+        Perform call to the other side of the channel to get its channel id
+        Each side is generating the channel id by itself so there is no way to identify a connection without this sync
+        '''
+        if self._other_channel_id is None:
```
[the remark this answers is not in the document]
Comment (asafc) [reply fastapi_websocket_rpc/rpc_channel.py:263]: super important @obsd

**Review by asafc — CHANGES_REQUESTED** [review]
Good work @obsd!

I am going to block until we fix the two criticial comments:
1) two methods with same name - `get_other_channel_id()`
2) forgot else block (Or's comment)

fixes should be really quick, almost there!

File: fastapi_websocket_rpc/rpc_channel.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (obsd) [reply fastapi_websocket_rpc/rpc_channel.py:263]: fixed both issues and forgot to push :|

File: fastapi_websocket_rpc/rpc_channel.py [hunk 3]
```diff
@@ -234,8 +247,24 @@ async def on_handler_event(self, handlers, *args, **kwargs):
...
+            other_channel_id = await self.other._get_channel_id_()
+            self._other_channel_id = other_channel_id.result if other_channel_id and other_channel_id.result else None
```
[the remark this answers is not in the document]
Comment (obsd) [reply fastapi_websocket_rpc/rpc_channel.py:265]: If I have here a None should I raise an error?

[the remark this answers is not in the document]
Comment (obsd) [reply fastapi_websocket_rpc/rpc_channel.py:270]: @asafc

**Review by asafc — APPROVED** [review]

**Pull request merged**